addons/survey/models/survey_survey.py

```diff
@@ -583,9 +583,32 @@ def _get_pages_or_questions(self, user_input):
             else:
                 result = self.question_and_page_ids.filtered(
                     lambda question: not question.is_page or not is_html_empty(question.description))
```

Comment:
This can be removed then, right?

Reply:
It does make sense to move that logic in the new method - it's now done
I also simplified a bit the assertEqual calls so they're more consistent and easier to understand
I also added a page without description in the test to cover that behavior in the method